index.d.ts

```diff
@@ -13,10 +13,29 @@ type FastifyCorsPlugin = FastifyPluginCallback<
   NonNullable<fastifyCors.FastifyCorsOptions> | fastifyCors.FastifyCorsOptionsDelegate
 >;
 
+type FastifyCorsHook =
+  | 'onRequest'
+  | 'preParsing'
+  | 'preValidation'
+  | 'preHandler'
+  | 'preSerialization'
+  | 'onSend'
+  | 'onError'
```

Comment:
`onError` doesn't seem to be correct. Why you only need `CORS` when it is error?

Reply:
I just chose the hooks which can modify the request before being sent. Anyway. Should it be removed?

Reply:
I would say yes, as `onError` hook may lead to unexpected behavior to the user.

Reply:
done